packages/plugins/experimental/plugin-transcription/src/capabilities/app-graph-builder.ts

```diff
@@ -3,28 +3,148 @@
+import { generateName } from '@dxos/display-name';
+import { getSchemaTypename } from '@dxos/echo-schema';
+import { invariant } from '@dxos/invariant';
+import { ClientCapabilities } from '@dxos/plugin-client';
+import { COMPANION_TYPE } from '@dxos/plugin-deck/types';
 import { createExtension, type Node } from '@dxos/plugin-graph';
+import { MeetingCapabilities, type CallState, type MediaState } from '@dxos/plugin-meeting';
+import { MeetingType } from '@dxos/plugin-meeting/types';
+import { buf } from '@dxos/protocols/buf';
+import {
+  TranscriptionPayloadSchema,
+  type TranscriptionPayload as TranscriptionPayloadProto,
+} from '@dxos/protocols/buf/dxos/edge/calls_pb';
+import { keyToFallback } from '@dxos/util';
 
+import { TranscriptionCapabilities } from './capabilities';
 import { TRANSCRIPTION_PLUGIN } from '../meta';
+import { TranscriptionManager } from '../transcriber';
 import { TranscriptionAction, TranscriptType } from '../types';
 
+// TODO(wittjosiah): Factor out.
+// TODO(wittjosiah): Can we stop using protobuf for this?
+type TranscriptionPayload = buf.MessageInitShape<typeof TranscriptionPayloadSchema>;
+const codec = {
+  encode: (message: TranscriptionPayload): Uint8Array => {
+    return buf.toBinary(TranscriptionPayloadSchema, buf.create(TranscriptionPayloadSchema, message));
+  },
+  decode: (message: Uint8Array): TranscriptionPayloadProto => {
+    return buf.fromBinary(TranscriptionPayloadSchema, message);
+  },
+};
+
+const getMeetingTranscript = async (
+  context: PluginsContext,
+  meeting: MeetingType,
+): Promise<TranscriptType | undefined> => {
+  const { dispatchPromise: dispatch } = context.requestCapability(Capabilities.IntentDispatcher);
+
+  const typename = getSchemaTypename(TranscriptType)!;
+  const existing = await meeting.artifacts[typename]?.load();
+  if (existing instanceof TranscriptType) {
+    return existing;
+  }
+
+  const space = getSpace(meeting);
+  invariant(space);
+  const { data } = await dispatch(createIntent(TranscriptionAction.Create, { spaceId: space.id }));
+  meeting.artifacts[typename] = makeRef(data!.object);
+  return data?.object;
+};
+
+// TODO(wittjosiah): Introduce a sound that plays when transcription is activated/deactivated.
 export default (context: PluginsContext) =>
   contributes(Capabilities.AppGraphBuilder, [
     createExtension({
-      id: `${TRANSCRIPTION_PLUGIN}/transcription`,
-      filter: (node): node is Node<TranscriptType> => node.data instanceof TranscriptType,
-      actions: () => [
-        {
-          id: `${TRANSCRIPTION_PLUGIN}/transcription`,
-          data: async () => {
-            const { dispatchPromise: dispatch } = context.requestCapability(Capabilities.IntentDispatcher);
-            await dispatch(createIntent(TranscriptionAction.Summarize));
+      id: `${TRANSCRIPTION_PLUGIN}/meeting-transcript`,
+      filter: (node): node is Node<MeetingType> => node.data instanceof MeetingType,
```

Comment:
Does `S.is(MeetingType)(node.data)` work?

Reply:
that would be a structural check, I'm assuming that this will change to `Echo.isInstanceOf(MeetingType, node.data)` once we finish migrating to the new apis for all the schemas